### ISSUE TITLE
Running with DevTools when coming back from popped-up window hangs WebdriverIO

### ISSUE THREAD

**Issue by gabriel-kohen-by** [webdriverio/webdriverio#6623]
[//]: # NOTE: This repository only maintains packages that are listed in the [Readme](https://github.com/webdriverio/webdriverio/blob/main/README.md#packages). Please make sure that your issue is directly caused by one of these packages and if not file an issue in the correct 3rd party package repository.

**Environment (please complete the following information):**
 - **WebdriverIO version:** [7.0.8]
 - **Mode:** [Testrunner]
 - **If WDIO Testrunner, running sync/async:** [sync]
 - **Node.js version:** [14.16.0]
 - **NPM version:** [6.14.11]
 - **Browser name and version:** [Chrome 89]
 - **Platform name and version:** [MacOS Catalina]
 - **Additional wdio packages used (if applicable):
   [
    "@wdio/appium-service": "^7.0.7",
    "@wdio/cli": "^7.1.2",
    "@wdio/devtools-service": "^7.2.3",
    "@wdio/local-runner": "^7.0.8",
    "@wdio/logger": "^7.0.0",
    "@wdio/mocha-framework": "^7.0.7",
    "@wdio/sauce-service": "^7.0.8",
    "@wdio/spec-reporter": "^7.2.1",
    "@wdio/sync": "^7.0.8",
    "ts-node": "^9.1.1",
    "typescript": "^3.9.7",
    "wdio-chromedriver-service": "^6.0.4",
    "wdio-geckodriver-service": "^1.1.0",
    "wdio-spec-reporter": "^0.1.5",
    "wdio-video-reporter": "^3.1.1"
   ]

**Config of WebdriverIO**
DevTools Specific:
```
import { baseConfig as config } from './wdio.shared.conf';
config.services.push('devtools');
exports.config = config;
```
Shared:
```
/* eslint-disable @typescript-eslint/no-unused-vars */
const debug = process.env.DEBUG;
require('expect-webdriverio');
const getLogger = require('@wdio/logger').default;
const log = getLogger('hooks');
const video = require('wdio-video-reporter');
if (
  !(
    process.env.FUNCTIONAL_TEST_USERNAME &&
    process.env.FUNCTIONAL_TEST_PASSWORD &&
    process.env.FUNCTIONAL_TEST_REALM &&
    process.env.FUNCTIONAL_TEST_IDP
  )
) {
  const error = 'Please provide functional user name, password, realm and IdP description';
  console.error(error);
  throw new Error(error);
}
const baseConfig: WebdriverIO.Config = {
  //
  // ====================
  // Runner Configuration
  // ====================
  //
  // WebdriverIO allows it to run your tests in arbitrary locations (e.g. locally or
  // on a remote machine).
  runner: 'local',
  //
  // ==================
  // Specify Test Files
  // ==================
  // Define which test specs should run. The pattern is relative to the directory
  // from which `wdio` was called. Notice that, if you are calling `wdio` from an
  // NPM script (see https://docs.npmjs.com/cli/run-script) then the current working
  // directory is where your package.json resides, so `wdio` will be called from there.
  //
  specs: ['./e2e/specs/**/test-popup.ts'],
  // Patterns to exclude.
  exclude: [
    // 'path/to/excluded/files'
  ],
  // ============
  // Capabilities
  // ============
  // Define your capabilities here. WebdriverIO can run multiple capabilities at the same
  // time. Depending on the number of capabilities, WebdriverIO launches several test
  // sessions. Within your capabilities you can overwrite the spec and exclude options in
  // order to group specific specs to a specific capability.
  //
  // First, you can define how many instances should be started at the same time. Let's
  // say you have 3 different capabilities (Chrome, Firefox, and Safari) and you have
  // set maxInstances to 1; wdio will spawn 3 processes. Therefore, if you have 10 spec
  // files and you set maxInstances to 10, all spec files will get tested at the same time
  // and 30 processes will get spawned. The property handles how many capabilities
  // from the same test should run tests.
  //
  maxInstances: debug ? 1 : 4,
  //
  // If you have trouble getting all important capabilities together, check out the
  // Sauce Labs platform configurator - a great tool to configure your capabilities:
  // https://docs.saucelabs.com/reference/platforms-configurator
  //
  capabilities: [
    {
      // maxInstances can get overwritten per capability. So if you have an in-house Selenium
      // grid with only 5 firefox instances available you can make sure that not more than
      // 5 instances get started at a time.
      // maxInstances: 5,
      //
      browserName: 'chrome',
      // If outputDir is provided WebdriverIO can capture driver session logs
      // it is possible to configure which logTypes to include/exclude.
      // excludeDriverLogs: ['*'], // pass '*' to exclude all driver session logs
      // excludeDriverLogs: ['bugreport', 'server'],
    },
  ],
  //
  // ===================
  // Test Configurations
  // ===================
  // Define all options that are relevant for the WebdriverIO instance here
  //
  // Level of logging verbosity: trace | debug | info | warn | error | silent
  logLevel: debug ? 'debug' : 'info',
  //
  // Set specific log levels per logger
  // loggers:
  // - webdriver, webdriverio
  // - @wdio/applitools-service, @wdio/browserstack-service, @wdio/devtools-service, @wdio/sauce-service
  // - @wdio/mocha-framework, @wdio/jasmine-framework
  // - @wdio/local-runner, @wdio/lambda-runner
  // - @wdio/sumologic-reporter
  // - @wdio/cli, @wdio/config, @wdio/sync, @wdio/utils
  // Level of logging verbosity: trace | debug | info | warn | error | silent
  // logLevels: {
  //     webdriver: 'info',
  //     '@wdio/applitools-service': 'info'
  // },
  outputDir: '.e2e/logs',
  //
  // If you only want to run your tests until a specific amount of tests have failed use
  // bail (default is 0 - don't bail, run all tests).
  bail: 0,
  //
  // Set a base URL in order to shorten url command calls. If your `url` parameter starts
  // with `/`, the base url gets prepended, not including the path portion of your baseUrl.
  // If your `url` parameter starts without a scheme or `/` (like `some/path`), the base url
  // gets prepended directly.
  baseUrl: process.env.LUMINATE_PORTAL_URL ?? 'http://localhost:7777/',
  //
  // Default timeout for all waitFor* commands.
  waitforTimeout: 5000,
  //
  // Default timeout in milliseconds for request
  // if browser driver or grid doesn't send response
  connectionRetryTimeout: 30000,
  //
  // Default request retries count
  connectionRetryCount: 9,
  //
  // Test runner services
  // Services take over a specific job you don't want to take care of. They enhance
  // your test setup with almost no effort. Unlike plugins, they don't add new
  // commands. Instead, they hook themselves up into the test process.
  services: [],
  autoCompileOpts: {
    autoCompile: true,
    tsNodeOpts: {
      transpileOnly: true,
      project: 'tsconfig.e2e.json',
    },
  },

  // Framework you want to run your specs with.
  // The following are supported: Mocha, Jasmine, and Cucumber
  // see also: https://webdriver.io/docs/frameworks.html
  //
  // Make sure you have the wdio adapter package for the specific framework installed
  // before running any tests.
  framework: 'mocha',
  //
  // The number of times to retry the entire specfile when it fails as a whole
  // specFileRetries: 1,
  //
  // Whether or not retried specfiles should be retried immediately or deferred to the end of the queue
  // specFileRetriesDeferred: false,
  //
  // Test reporter for stdout.
  // The only one supported by default is 'dot'
  // see also: https://webdriver.io/docs/dot-reporter.html
  reporters: [
    [
      video,
      {
        saveAllVideos: false, // If true, also saves videos for successful test cases
        videoSlowdownMultiplier: 3, // Higher to get slower videos, lower for faster videos [Value 1-100]
      },
    ],
    [
      'allure',
      {
        outputDir: '.e2e/reports/allure-raw',
        disableWebdriverStepsReporting: true,
        disableWebdriverScreenshotsReporting: true,
      },
    ],
  ],
  mochaOpts: {
    timeout: debug ? 60000 * 60 * 24 : 60000,
  },

  //
  // =====
  // Hooks
  // =====
  // WebdriverIO provides several hooks you can use to interfere with the test process in order to enhance
  // it and to build services around it. You can either apply a single function or an array of
  // methods to it. If one of them returns with a promise, WebdriverIO will wait until that promise got
  // resolved to continue.
  /**
   * Gets executed once before all workers get launched.
   * @param {Object} config wdio configuration object
   * @param {Array.<Object>} capabilities list of capabilities details
   */
  // onPrepare: function (config, capabilities) {
  // },
  /**
   * Gets executed before a worker process is spawned and can be used to initialise specific service
   * for that worker as well as modify runtime environments in an async fashion.
   * @param  {String} cid      capability id (e.g 0-0)
   * @param  {[type]} caps     object containing capabilities for session that will be spawn in the worker
   * @param  {[type]} specs    specs to be run in the worker process
   * @param  {[type]} args     object that will be merged with the main configuration once worker is initialised
   * @param  {[type]} execArgv list of string arguments passed to the worker process
   */
  // onWorkerStart: function (cid, caps, specs, args, execArgv) {
  // },
  /**
   * Gets executed just before initialising the webdriver session and test framework. It allows you
   * to manipulate configurations depending on the capability or spec.
   * @param {Object} config wdio configuration object
   * @param {Array.<Object>} capabilities list of capabilities details
   * @param {Array.<String>} specs List of spec file paths that are to be run
   */
  // beforeSession: function (config, capabilities, specs) {
  // },
  /**
   * Gets executed before test execution begins. At this point you can access to all global
   * variables like `browser`. It is the perfect place to define custom commands.
   * @param {Array.<Object>} capabilities list of capabilities details
   * @param {Array.<String>} specs List of spec file paths that are to be run
   */
  before: function (capabilities, specs) {
    browser.addCommand('customConfig', function () {
      return {
        text: 'Hi Custom Command',
      };
    });
  },
  /**
   * Runs before a WebdriverIO command gets executed.
   * @param {String} commandName hook command name
   * @param {Array} args arguments that command would receive
   */
  // beforeCommand: function (commandName, args) {
  // },
  /**
   * Hook that gets executed before the suite starts
   * @param {Object} suite suite details
   */
  beforeSuite: function (suite) {
    log.debug(`Suite "${suite.fullTitle}" from file "${suite.file}" starts`);
  },
  /**
   * Function to be executed before a test (in Mocha/Jasmine) starts.
   */
  beforeTest: function (test, context) {
    log.debug(`Test "${test.title}" starts`);
  },
  /**
   * Hook that gets executed _before_ a hook within the suite starts (e.g. runs before calling
   * beforeEach in Mocha)
   */
  // beforeHook: function (test, context) {
  // },
  /**
   * Hook that gets executed _after_ a hook within the suite starts (e.g. runs after calling
   * afterEach in Mocha)
   */
  // afterHook: function (test, context, { error, result, duration, passed, retries }) {
  // },
  /**
   * Function to be executed after a test (in Mocha/Jasmine).
   */
  afterTest: function (test, context, { error, result, duration, passed, retries }) {
    log.debug(`Test "${test.title}" finished`);
  },

  /**
   * Hook that gets executed after the suite has ended
   * @param {Object} suite suite details
   */
  afterSuite: function (suite) {
    log.debug(`Suite "${suite.fullTitle}" from file "${suite.file}" finished`);
  },
  /**
   * Runs after a WebdriverIO command gets executed
   * @param {String} commandName hook command name
   * @param {Array} args arguments that command would receive
   * @param {Number} result 0 - command success, 1 - command error
   * @param {Object} error error object if any
   */
  // afterCommand: function (commandName, args, result, error) {
  // },
  /**
   * Gets executed after all tests are done. You still have access to all global variables from
   * the test.
   * @param {Number} result 0 - test pass, 1 - test fail
   * @param {Array.<Object>} capabilities list of capabilities details
   * @param {Array.<String>} specs List of spec file paths that ran
   */
  // after: function (result, capabilities, specs) {
  // },
  /**
   * Gets executed right after terminating the webdriver session.
   * @param {Object} config wdio configuration object
   * @param {Array.<Object>} capabilities list of capabilities details
   * @param {Array.<String>} specs List of spec file paths that ran
   */
  // afterSession: function (config, capabilities, specs) {
  // },
  /**
   * Gets executed after all workers got shut down and the process is about to exit. An error
   * thrown in the onComplete hook will result in the test run failing.
   * @param {Object} exitCode 0 - success, 1 - fail
   * @param {Object} config wdio configuration object
   * @param {Array.<Object>} capabilities list of capabilities details
   * @param {<Object>} results object containing test results
   */
  // onComplete: function(exitCode, config, capabilities, results) {
  // },
  /**
   * Gets executed when a refresh happens.
   * @param {String} oldSessionId session ID of the old session
   * @param {String} newSessionId session ID of the new session
   */
  // onReload: function(oldSessionId, newSessionId) {
  // }
};

exports.baseConfig = baseConfig;
```

**Describe the bug**
When WDIO goes from a parent page to a popup window, the popup window has a button that closes it, WDIO clicks the button. At this point we cannot get windows handles or switch to a window. This works fine with ChromeDriver or SauceLabs

**To Reproduce**
Create two HTML pages:
1. Parent with a button to open a child.
2. A child with a button to close itself
3. Run the example:
4. parent.html
```
<html>

<head>
    <title>Parcel Sandbox</title>
    <meta charset="UTF-8" />

</head>

<body>
<div id="app"></div>


<h1>Hello WebdriverIO!</h1>
<div>
    Look
    <button id="parentButton">Open Popup</button>
    <button id="otherButton">Try Me</button>
</div>
<div id="message">^^^^^^^^^^^^^Try to click the one above</div>
<script>
    document.getElementById("parentButton").addEventListener("click", () => {
        console.log("Test");
        window.open("./child.html", "TestPopup","resizable=no,scrollbars=no,status=no,width=500,height=300");
    });
    document.getElementById("otherButton").addEventListener("click", () => {
        document.getElementById("message").innerText="I've been clicked"
        setTimeout(function(){
            document.getElementById("message").innerText="^^^^^^^^^^^^^Try to click the one above"
        },5000)
    });
</script>
</body>

</html>
```
child.html
```
<html>

<head>
    <title>Sample Login</title>
    <meta charset="UTF-8" />
</head>

<body>
<div id="app"></div>
<h1>Hello Login!</h1>
<div>
    <button id="closeButton" >Close me</button>
</div>
<script>
    document.getElementById("closeButton").addEventListener("click", () => {
        window.close();
    });
</script>
</body>

</html>
```

[Include code or an example repository that can easily be set up]

**Expected behavior**
WDIO should not hang on the commands above(switchToWindow/getWindowsHandles). 

**Log**
Logs and files mentioned above are in the following gist:
https://gist.github.com/gabriel-kohen-by/7018694007aaf28cacc2f57747e663c5

**Comment by erwinheitzman** [comment]
Confirmed, code hangs when using the supplied examples and the simplified code found below:
```js
const { remote } = require('webdriverio');

let browser;

(async () => {
	browser = await remote({
		automationProtocol: 'devtools',
		capabilities: {
				browserName: 'chrome',
				logLevel: 'debug',
				'wdio:devtoolsOptions': {
						headless: false,
				}
		}
	})

	await browser.url('file:///D:/dev/deletemetoo/parent.html');
	
	const parentButton = await browser.$('#parentButton')
	await parentButton.click();

	await browser.switchWindow('Sample Login');

	const closeButton = await browser.$('#closeButton')
	await closeButton.click();

	await browser.deleteSession()
})().catch(async (e) => {
	console.error(e)
	await browser.deleteSession()
})
```

**Comment by gabriel-kohen-by** [comment]
FYI: @rohit-pant-by

**Comment by christian-bromann** [comment]
fixed in #6924
